### DISCUSSION TITLE
Function get_full_dimensional_polytope()

### DISCUSSION THREAD

**Discussion by imitator19** [GeomScale/volesti#213] [Q&A]
Hello, everyone, I find that volesti has the function "get_full_dimensional_polytope", with description "Given a low-dimensional H-polytope, restrict it to its null space". However, this function is not included in the document. I am confused about its underlying algorithm, as well as the attributes of H-polytope class used in this function: Aeq and beq and function output. Could anyone provide some introduction on this function? Thanks a lot!

**Comment by vaithak** [comment]
@imitator19 Can you point out where is that function defined/documented? I did a search on the codebase in the develop branch and couldn't find any references to this function.

**Reply by imitator19** [reply]
I am currently using the [v1.1.3] of volesti (https://github.com/GeomScale/volesti/releases/tag/v1.1.3) . There are one ".cpp" file, one ".hpp" file and one ".R" file, named by "get_full_dimensional_polytope" in the codebase. The brief introduction of this function is placed in the  ".R" file, and can be accessed with the command "help("get_full_dimensional_polytope")" in Rstudio.
Thanks!

**Reply by vissarion** [reply]
@imitator19 please use links to the code, then it is much simpler for others to follow the discussion. 

For example, this is the `C++` file that contains the function `get_full_dimensional_polytope` https://github.com/GeomScale/volesti/blob/c9cc0ecb5e7e6d62fb05f8cef5e509f054eceab8/include/preprocess/get_full_dimensional_polytope.hpp 

The file was removed by this PR https://github.com/GeomScale/volesti/pull/194 I guess because it was not used anywhere, @papachristoumarios could you help us here?

**Reply by imitator19** [reply]
Thank you for your reply. I find that with this function, non-full dimensional polytope can be sampled. In addition to the "A" and "b" parameter in the polytope equation"Ax<=b", "Aeq" and "beq" can be assigned to the polytope object attributes to describe "Aeq*x=beq". 
I think this is a useful feature to sample non-full dimensional polytope, making volesti more powerful. 
An introduction to the underlying algorithm will be much appreciated.
